File: 0448-find-all-numbers-disappeared-in-an-array/0448-find-all-numbers-disappeared-in-an-array.py
class Solution:
    def findDisappearedNumbers(self, nums: List[int]) -> List[int]:
        being = 0
        while being != len(nums):
            pos = nums[being]-1
            if nums[being] != nums[pos]:
                nums[being], nums[pos] = nums[pos], nums[being]
            else:
                being += 1
        result = []
        for i in range(len(nums)):
            if nums[i] != i+1:
                result.append(i+1)
        return result

File: 0448-find-all-numbers-disappeared-in-an-array/0448-find-all-numbers-disappeared-in-an-array.java
class Solution {
    public List<Integer> findDisappearedNumbers(int[] nums) {
        int i = 0;
        while (i < nums.length) {
            int pos = nums[i] - 1;
            if (nums[pos] != nums[i]) {
                int tmp = nums[pos];
                nums[pos] = nums[i];
                nums[i] = tmp;
            } else {
                i++;
            }
        }
        List<Integer> res = new ArrayList<>();
        for (int j = 0; j < nums.length; j++) {
            if (nums[j] != j+1) {
                res.add(j+1);
            }
        }
        return res;
    }
}